File: Categories/Array/853_carFleet.py
from typing import *


class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        n = len(position)

        pairs = []
        for pos, spd in zip(position, speed):
            pairs.append( (pos, spd) )
        pairs.sort(key = lambda x: x[0])

        ret = 1
        prev = n - 1
        for idx in range(n - 2, -1, -1):
            t_prev = (target - pairs[prev][0]) / pairs[prev][1]
            t_cur = (target - pairs[idx][0]) / pairs[idx][1]

            if t_cur > t_prev:
                prev = idx
                ret += 1
        
        return ret

File: Categories/Array/853_carFleet.cs
public class Solution {
    public int CarFleet(int target, int[] position, int[] speed) {
        int n = position.Length;
        List<Tuple<int,int>> pairs = new List<Tuple<int, int>>();

        for (int idx = 0; idx < n; idx++) {
            pairs.Add(new Tuple<int,int>(position[idx], speed[idx]));
        }

        pairs = pairs.OrderBy(p => p.Item1).ToList();

        int prev = n - 1;
        int ret = 1;
        for (int idx = n - 2; idx >= 0; idx--) {
            double t_prev = (double)(target - pairs[prev].Item1) / pairs[prev].Item2;
            double t_cur = (double)(target - pairs[idx].Item1) / pairs[idx].Item2;
            if (t_cur > t_prev) {
                ret += 1;
                prev = idx;
            }
        }

        return ret;
    }
}